src/traces/pie/plot.js

```diff
@@ -164,36 +169,39 @@ function plot(gd, cdModule) {
                         if(textPosition === 'auto' && transform.scale < 1) {
-                            sliceText.call(Drawing.font, trace.outsidetextfont);
-                            if(trace.outsidetextfont.family !== trace.insidetextfont.family ||
-                                    trace.outsidetextfont.size !== trace.insidetextfont.size) {
+                            var newFont = Lib.extendFlat({}, trace.outsidetextfont, {});
+                            newFont.size = Math.max(newFont.size, fullLayout.uniformtext.minsize || 0);
+
+                            sliceText.call(Drawing.font, newFont);
+                            if(newFont.family !== font.family || newFont.size !== font.size) {
+                                // recompute bounding box
                                 textBB = Drawing.bBox(sliceText.node());
                             }
                             transform = transformOutsideText(textBB, pt);
                         }
                     }
 
-                    var translateX = cx + pt.pxmid[0] * transform.rCenter + (transform.x || 0);
-                    var translateY = cy + pt.pxmid[1] * transform.rCenter + (transform.y || 0);
+                    var pxtxt = pt.pxtxt || pt.pxmid;
+                    transform.targetX = cx + pxtxt[0] * transform.rCenter + (transform.x || 0);
+                    transform.targetY = cy + pxtxt[1] * transform.rCenter + (transform.y || 0);
+                    computeTransform(transform, textBB);
 
                     // save some stuff to use later ensure no labels overlap
                     if(transform.outside) {
-                        pt.yLabelMin = translateY - textBB.height / 2;
-                        pt.yLabelMid = translateY;
-                        pt.yLabelMax = translateY + textBB.height / 2;
+                        var targetY = transform.targetY;
+                        pt.yLabelMin = targetY - textBB.height / 2;
+                        pt.yLabelMid = targetY;
+                        pt.yLabelMax = targetY + textBB.height / 2;
                         pt.labelExtraX = 0;
                         pt.labelExtraY = 0;
                         hasOutsideText = true;
                     }
 
-                    sliceText.attr('transform',
-                        'translate(' + translateX + ',' + translateY + ')' +
-                        (transform.scale < 1 ? ('scale(' + transform.scale + ')') : '') +
-                        (transform.rotate ? ('rotate(' + transform.rotate + ')') : '') +
-                        'translate(' +
-                            (-(textBB.left + textBB.right) / 2) + ',' +
-                            (-(textBB.top + textBB.bottom) / 2) +
-                        ')');
+                    transform.fontSize = font.size;
+                    recordMinTextSize(trace.type, transform, fullLayout);
+                    cd[i].transform = transform;
```

Comment:
Why do we need to stash the transform info in the calcdata?

Reply:
In `resizeText` we need to have access to translate, scale and rotation raw values without parsing the style of the element.

Reply:
See https://github.com/plotly/plotly.js/pull/4420/files#r357786862

Reply:
Ok - I get that's fine though I might be hard to keep that thing in-sync in the future. Closing.